### PR TITLE
ci: add userstyle linter

### PR DESCRIPTION
**Pull request by nekowinston**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Lint userstyles
```
Comment (sgoudham) [inline .github/workflows/lint.yml:1]:
```suggestion
name: Userstyle Linter
```

This matches with the other workflow names, however, we might want to change the others to match with this PR instead.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
```
Comment (sgoudham) [inline .github/workflows/lint.yml:10]:
```suggestion
    runs-on: ubuntu-latest
    name: Lint
```

Matching case with the other workflows

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Userstyle Linter
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  lint:
+	name: Lint
```
Comment (sgoudham) [inline .github/workflows/lint.yml:10]:
```suggestion
    name: Lint
```

File: src/lint/lint.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      core.error(err, { file: entry.relative });
+    }
+
+    const assert = assertions(repo);
+
+    Object.entries(assert).forEach(([k, v]) => {
+      const defacto = metadata[k];
+      if (defacto !== v) {
+        const line = css.split("\n").findIndex((line) => line.includes(k)) + 1;
+        core.warning(`Metadata ${k} should be ${v} but is ${defacto}`, {
```
Comment (uncenter) [inline src/lint/lint.ts:42]: Do warnings support markdown? I would prefer something like this (template values wrapped in inline code backticks, punctuation):
```js
`Metadata key \`${k}\` should be \`${v}\` but is \`${defacto}\`.`
```

> Metadata key `preprocessor` should be `less` but is `stylus`.

**Review by uncenter — APPROVED** [review]
LGTM.

**Pull request closed without merging**